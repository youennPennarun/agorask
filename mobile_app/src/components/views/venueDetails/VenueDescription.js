import React, {PropTypes} from 'react';
import {View, Text, StyleSheet, Dimensions} from 'react-native';
import Icon from 'react-native-vector-icons/Entypo';

const {width} = Dimensions.get('window');

function getIcon(amenity) {
  if (amenity === 'pub') return 'drink';
  return 'globe';
}
function renderAddress(formattedAddress) {
  if (typeof formattedAddress === 'string') {
    return (<Text>{formattedAddress}</Text>);
  } else if (formattedAddress instanceof Array) {
    return (
      <View>{formattedAddress.map((line, key) => <Text key={key}>{line}</Text>)}</View>
    );
  }
  return <View />;
}

export default function VenueDescription(props) {
  const {name, address, categories = [], website, contact = {formattedPhone: ''}} = props.venue;
  const categoryName = (categories.length) ? categories[0].name : '';
  return (
    <View style={styles.block}>
      <View style={styles.titleContainer} >
        <Icon name={getIcon()}
          style={styles.icon}
          size={16}
          color='blue' />
        <Text style={styles.venueName}>{name}</Text>
      </View>
      {renderAddress(address.formatted)}
      <Text>{categoryName}</Text>
      <Text>{website}</Text>
      {(contact.formattedPhone) ? (
        <Text><Icon name={'mobile'}
          style={styles.icon}
          size={16}
          color='blue' />{contact.formattedPhone}
        </Text>) : null
      }
    </View>
  );
}

const styles = StyleSheet.create({
  block: {
    width,
    backgroundColor: 'white',
    marginTop: 5,
    paddingTop: 10,
    paddingLeft: 10,
    paddingRight: 10,
    paddingBottom: 10,
  },
  icon: {
    marginRight: 15,
  },
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  venueName: {
    fontSize: 20,
  },
});

VenueDescription.propTypes = {
  venue: PropTypes.object.isRequired,
};
