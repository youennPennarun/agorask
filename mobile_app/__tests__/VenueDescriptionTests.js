/* global jest, expect, describe, it */

import 'react-native';
import React from 'react';

import VenueDescription from '../src/components/views/venueDetails/VenueDescription';

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

const venueTest1 = {
  name: 'Test venue',
  address: {
    formatted: [
      '2 univ street',
      '5200 Belfast',
      'UK',
    ],
  },
  categories: [],
  contact: {
    formattedPhone: '(060) 25-65',
  },
};

const venueTest2 = {
  name: 'Test venue',
  address: {
    formatted: '2 univ street 5200 Belfast UK',
  },
  categories: [],
  contact: {
    formattedPhone: '(060) 25-65',
  },
};

const venueTest3 = {
  name: 'Test venue',
  address: {
  },
  categories: [],
  contact: {
    formattedPhone: '(060) 25-65',
  },
};

it('Test with formatted address', () => {
  const wrapper = shallow(
    <VenueDescription venue={venueTest1} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('Test with address as a single string', () => {
  const wrapper = shallow(
    <VenueDescription venue={venueTest2} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});


it('Test without address', () => {
  const wrapper = shallow(
    <VenueDescription venue={venueTest3} />
  );
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});


