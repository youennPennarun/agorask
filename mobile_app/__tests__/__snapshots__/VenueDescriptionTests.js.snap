exports[`test Test with address as a single string 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "elevation": 2,
      "marginTop": 5,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }>
    <Icon
      allowFontScaling={false}
      color="blue"
      name="globe"
      size={16}
      style={
        Object {
          "marginRight": 15,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#212121",
          "fontSize": 21,
        }
      }>
      Test venue
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    2 univ street 5200 Belfast UK
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    <Icon
      allowFontScaling={false}
      color="blue"
      name="mobile"
      size={16}
      style={
        Object {
          "marginRight": 15,
        }
      } />
    (060) 25-65
  </Text>
</View>
`;

exports[`test Test with formatted address 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "elevation": 2,
      "marginTop": 5,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }>
    <Icon
      allowFontScaling={false}
      color="blue"
      name="globe"
      size={16}
      style={
        Object {
          "marginRight": 15,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#212121",
          "fontSize": 21,
        }
      }>
      Test venue
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      2 univ street
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      5200 Belfast
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      UK
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    <Icon
      allowFontScaling={false}
      color="blue"
      name="mobile"
      size={16}
      style={
        Object {
          "marginRight": 15,
        }
      } />
    (060) 25-65
  </Text>
</View>
`;

exports[`test Test without address 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "elevation": 2,
      "marginTop": 5,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": 750,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }>
    <Icon
      allowFontScaling={false}
      color="blue"
      name="globe"
      size={16}
      style={
        Object {
          "marginRight": 15,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#212121",
          "fontSize": 21,
        }
      }>
      Test venue
    </Text>
  </View>
  <View />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail" />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
    <Icon
      allowFontScaling={false}
      color="blue"
      name="mobile"
      size={16}
      style={
        Object {
          "marginRight": 15,
        }
      } />
    (060) 25-65
  </Text>
</View>
`;
